SDP-39 Add test for CreateEvent

dadlex opened the pull request:
Add test to check that event is created.
Also change composition back to inheritance. At least until we decide to change pravegaStorageDriver methods' types.

The only question left: how to get size of the stream to check that its size == dataItem.size(). Until that we only check operation status and equality of dataItem and created task.

dadlex committed:
Add test for CreateEvent

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Add test to check that event is created.
Also change composition back to inheritance. At least until we decide to change pravegaStorageDriver methods' types.

Signed-off-by: Alexander <alex15051997@yandex.com>

dadlex committed:
Add test for CreateEvent

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Implement ByteBufferSerializer that transforms ByteBuffer into Integer, showing how many bytes we've read.
Improve eventReadTest, so that we read the event we've out in the stream and check its size.

Signed-off-by: Alexander <alex15051997@yandex.com>

dadlex commented:
Pay attention to ByteBufferSerializer. I'm not sure it's doing what we want. But that's what I understood from our conversation.

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/io/ByteBufferSerializer.java:
```diff
@@ -0,0 +1,44 @@
+import java.nio.ByteBuffer;
+
+public class ByteBufferSerializer
+implements Serializer<Integer>, Serializable {
```
Looks to be a kind of bad idea. Let's use a deserializer which just returns the received byte buffer. Then we should just estimate the byte count in the buffer and forget it.

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/io/ByteBufferSerializer.java:
```diff
@@ -0,0 +1,44 @@
+		}
+		//as we do a flip in serialize buffer method after filling a buffer
+		//we don't need it here
+		int bytesDone = 0;
```
It should be just
`final int bytesDone = serializedValue.remaining();`
instead of loop

dadlex committed:
Add test for CreateEvent

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Change ByteBufferSerializer. Now it just returns ByteBuffer. We can get info about read bytes by using remaining() function.

Change eventReadTest according to the new serializer.

Signed-off-by: Alexander <alex15051997@yandex.com>
